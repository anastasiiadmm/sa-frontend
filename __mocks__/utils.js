import * as reduxHooks from 'react-redux';

jest.mock('../src/redux/store');
jest.mock('react-redux');

export const mockedUseSelectors = jest.spyOn(reduxHooks, 'useSelector');
export const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch');
