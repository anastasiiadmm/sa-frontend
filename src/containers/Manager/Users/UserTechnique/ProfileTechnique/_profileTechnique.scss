.ProfileTechnique {
  &__table {
    padding: 16px;
    background: #ffffff;
  }

  &__arrow-left {
    padding: 0 10px;
    cursor: pointer;
    width: 35px;
  }

  &__header {
    margin-bottom: 33px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__header-title {
    display: flex;
    align-items: baseline;
  }

  &__title {
    display: flex;
    font-weight: 400 !important;
  }

  &__subtitle {
    margin: 0;
    padding-left: 7px;
    padding-right: 7px;
    font-weight: 700 !important;
  }

  &__technique-profile-info {
    display: flex;
    gap: 37px;
  }

  &__form-block {
    display: flex;
    gap: 13px;
    > div {
      width: 100%;
    }
  }

  &__profile-title {
    margin-top: 0;
  }

  :where(.css-dev-only-do-not-override-sk7ap8).ant-form-vertical .ant-form-item-label, :where(.css-dev-only-do-not-override-sk7ap8).ant-col-24.ant-form-item-label, :where(.css-dev-only-do-not-override-sk7ap8).ant-col-xl-24.ant-form-item-label {
    padding: 0;
  }

  :where(.css-dev-only-do-not-override-sk7ap8).ant-input {
    padding: 0;
  }
}