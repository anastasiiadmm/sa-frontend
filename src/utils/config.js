export const apiURL = '';

export const defaultError = { detail: 'На сервере что-то пошло не так' };
