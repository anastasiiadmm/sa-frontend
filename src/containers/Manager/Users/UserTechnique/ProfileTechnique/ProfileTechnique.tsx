import { EyeOutlined } from '@ant-design/icons';
import { Button, Form, Image, Table, Typography } from 'antd';
import { ColumnsType, TableProps } from 'antd/es/table';
import bem from 'easy-bem';
import React from 'react';
import { Link } from 'react-router-dom';

import 'containers/Manager/Users/UserTechnique/ProfileTechnique/_profileTechnique.scss';
import arrowLeft from 'assets/images/icons/arrow-left.svg';
import technique from 'assets/images/image-profile-technique.jpg';
import FormField from 'components/FormField/FormField';

const { Title } = Typography;

interface DataType {
  key: React.Key;
  fields: string;
  area: string;
}

const ProfileTechnique = () => {
  const b = bem('ProfileTechnique');

  const columns: ColumnsType<DataType> = [
    {
      title: 'Поля техники',
      dataIndex: 'fields',
      width: '40%',
      sorter: true,
      fixed: 'left',
    },
    {
      title: 'Площадь поля',
      dataIndex: 'area',
      filterSearch: true,
      width: '40%',
      sorter: true,
    },
    {
      dataIndex: 'profile',
      filterSearch: true,
      width: '20%',
      render: () => (
        <div style={{ display: 'flex', gap: 37 }}>
          <Link className={b('profile-link')} to='/'>
            Просмотр на карте
          </Link>
        </div>
      ),
    },
  ];

  const data: DataType[] = [
    {
      key: '1',
      fields: 'Поле№1',
      area: '120 га',
    },
    {
      key: '2',
      fields: 'Поле№2',
      area: '120 га',
    },
    {
      key: '3',
      fields: 'Поле№3',
      area: '120 га',
    },
  ];

  const onChange: TableProps<DataType>['onChange'] = (pagination, filters, sorter, extra) => {};

  return (
    <div className={b()}>
      <div className={b('table')}>
        <div className={b('header')}>
          <div className={b('header-title')}>
            <Link to='/user-technique'>
              <img className={b('arrow-left')} src={arrowLeft} alt='arrow' />
            </Link>
            <Title level={3} className={b('title')}>
              Профиль техники - <p className={b('subtitle')}> AVP123344 </p> - John Deer
            </Title>
          </div>

          <div>
            <Button type='link' icon={<EyeOutlined />} size='large'>
              Техника на карте
            </Button>
          </div>
        </div>

        <div className={b('technique-profile-info')}>
          <Image src={technique} width={242} />
          <div className={b('profile-info')}>
            <Form autoComplete='off' layout='vertical'>
              <Title level={5} className={b('profile-title')}>
                Информация о технике
              </Title>
              <div className={b('form-block')}>
                <FormField
                  readOnly
                  id='technique_name_id'
                  label='Название техники'
                  name='technique_name'
                  placeholder='Название техники'
                />

                <FormField
                  readOnly
                  id='technique_number_id'
                  label='Гос номер'
                  name='technique_number'
                  placeholder='Гос номер'
                />

                <FormField
                  readOnly
                  id='code_id'
                  label='VIN код'
                  name='code'
                  placeholder='VIN код'
                />
              </div>
              <Title level={5} className={b('profile-title')}>
                Информация о механизаторе
              </Title>
              <div className={b('form-block')}>
                <FormField
                  readOnly
                  id='first_name_id'
                  label='Фамилия'
                  name='first_name'
                  placeholder='Фамилия'
                />

                <FormField
                  readOnly
                  id='last_name_id'
                  label='Имя'
                  name='last_name'
                  placeholder='Имя'
                />

                <FormField
                  readOnly
                  id='surname_id'
                  label='Отчество'
                  name='surname'
                  placeholder='Отчество'
                />
              </div>
            </Form>
          </div>
        </div>

        <Table
          scroll={{
            x: 950,
          }}
          columns={columns}
          dataSource={data}
          onChange={onChange}
        />
      </div>
    </div>
  );
};

export default ProfileTechnique;
