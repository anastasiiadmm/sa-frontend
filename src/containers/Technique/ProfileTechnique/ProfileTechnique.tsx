import { EyeOutlined } from '@ant-design/icons';
import { Button, Form, Image, Typography } from 'antd';
import { ColumnsType } from 'antd/es/table';
import bem from 'easy-bem';
import React, { useEffect, useState } from 'react';
import { Link, useParams } from 'react-router-dom';

import arrowLeft from 'assets/images/icons/arrow-left.svg';
import FormField from 'components/FormField/FormField';
import TableComponent from 'components/TableComponent/TableComponent';
import { companiesSelector, fetchUserVehicleInfo } from 'redux/companies/companiesSlice';
import { useAppDispatch, useAppSelector } from 'redux/hooks';
import { userVehicleInfo } from 'types';
import 'containers/Technique/ProfileTechnique/_profileTechnique.scss';

const { Title } = Typography;

const ProfileTechnique = () => {
  const b = bem('ProfileTechnique');
  const dispatch = useAppDispatch();
  const { userId, vehicleId } = useParams() as { userId: string; vehicleId: string };
  const { userVehicleInfo, userVehicleInfoLoading } = useAppSelector(companiesSelector);
  const [form] = Form.useForm();
  const [state, setState] = useState<userVehicleInfo[]>([]);

  useEffect(() => {
    dispatch(fetchUserVehicleInfo({ userId, vehicleId }));
  }, [dispatch]);

  useEffect(() => {
    if (userVehicleInfo) {
      setState((prevState) => [...prevState, userVehicleInfo]);
    }
  }, [userVehicleInfo]);

  useEffect(() => {
    if (state) {
      form.setFieldsValue({
        technique_name: `${state[0]?.last_name} ${state[0]?.first_name?.charAt(
          0,
        )}. ${state[0]?.middle_name?.charAt(0)}.`,
        state_number: state[0]?.state_number,
        vin_code: state[0]?.vin_code,
        last_name: state[0]?.last_name,
        first_name: state[0]?.first_name,
        middle_name: state[0]?.middle_name,
      });
    }
  }, [state]);

  const columns: ColumnsType<userVehicleInfo> = [
    {
      key: 'processing_data',
      title: 'Поля техники',
      dataIndex: 'fields',
      width: '18%',
      fixed: 'left',
      render: (text: string, record) => {
        return <p className={b('name-column-style')}>{record?.processing_data[0]?.field_name}</p>;
      },
    },
    {
      key: 'description',
      title: 'Инструмент',
      dataIndex: 'description',
      filterSearch: true,
      width: '17%',
    },
    {
      key: 'toolsWidth',
      title: 'Обрабатываемая ширина',
      dataIndex: 'description',
      filterSearch: true,
      width: '24%',
      render: (text: string, record) => {
        return (
          <p className={b('name-column-style')}>
            {record?.processing_data[0]?.attachments?.toolsWidth} м
          </p>
        );
      },
    },
    {
      key: 'skipOverlap',
      title: 'Ширина перекрытия',
      dataIndex: 'description',
      filterSearch: true,
      width: '17%',
      render: (text: string, record) => {
        return (
          <p className={b('name-column-style')}>
            {record?.processing_data[0]?.attachments?.skipOverlap} м
          </p>
        );
      },
    },
    {
      key: 'toolsWidthResult',
      title: 'Итоговая ширина',
      dataIndex: 'description',
      filterSearch: true,
      width: '15%',
      render: (text: string, record) => {
        return (
          <p className={b('name-column-style')}>
            {record?.processing_data[0]?.attachments?.toolsWidthResult} м
          </p>
        );
      },
    },
    {
      key: 'leftRight',
      title: 'Смещение',
      dataIndex: 'description',
      filterSearch: true,
      width: '15%',
      render: (text: string, record) => {
        return (
          <p className={b('name-column-style')}>
            {record?.processing_data[0]?.attachments?.leftRight} м
          </p>
        );
      },
    },
    {
      key: 'work_area',
      title: 'Обрабатываемая площадь',
      dataIndex: 'description',
      filterSearch: true,
      width: '21%',
      render: (text: string, record) => {
        return <p className={b('name-column-style')}>{record?.processing_data[0]?.work_area} га</p>;
      },
    },
    {
      dataIndex: 'profile',
      filterSearch: true,
      width: '23%',
      render: () => (
        <div style={{ display: 'flex', gap: 37 }}>
          <Link className={b('profile-link')} to='/open-map'>
            Просмотр на карте
          </Link>
        </div>
      ),
    },
  ];

  return (
    <div className={b()}>
      <div className={b('table')}>
        <div className={b('header')}>
          <div className={b('header-title')}>
            <Link to={`/user-technique/${userId}`}>
              <img className={b('arrow-left')} src={arrowLeft} alt='arrow' />
            </Link>
            <Title level={3} className={b('title')}>
              Профиль техники - <p className={b('subtitle')}> {state[0]?.vin_code} </p> -{' '}
              {`${state[0]?.last_name} ${state[0]?.first_name?.charAt(
                0,
              )}. ${state[0]?.middle_name?.charAt(0)}.`}
            </Title>
          </div>

          <div>
            <Link to='/open-map'>
              <Button type='link' icon={<EyeOutlined />} size='large'>
                Техника на карте
              </Button>
            </Link>
          </div>
        </div>

        <div className={b('technique-profile-info')}>
          <Image
            src={`https://agri.ltestl.com${state[0]?.image}`}
            width={242}
            style={{ borderRadius: 4 }}
          />
          <div className={b('profile-info')}>
            <Form form={form} initialValues={state[0]} layout='vertical'>
              <Title level={5} className={b('profile-title')}>
                Информация о технике
              </Title>
              <div className={b('form-block')}>
                <FormField
                  readOnly
                  id='technique_name_id'
                  label='Название техники'
                  name='technique_name'
                  placeholder='Название техники'
                  defaultValue={state[0]?.vin_code}
                />

                <FormField
                  readOnly
                  id='state_number_id'
                  label='Гос номер'
                  name='state_number'
                  placeholder='Гос номер'
                />

                <FormField
                  readOnly
                  id='vin_code_id'
                  label='VIN код'
                  name='vin_code'
                  placeholder='VIN код'
                />
              </div>
              <Title level={5} className={b('profile-title')}>
                Информация о механизаторе
              </Title>
              <div className={b('form-block')}>
                <FormField
                  readOnly
                  id='last_name_id'
                  label='Фамилия'
                  name='last_name'
                  placeholder='Фамилия'
                />

                <FormField
                  readOnly
                  id='first_name_id'
                  label='Имя'
                  name='first_name'
                  placeholder='Имя'
                />

                <FormField
                  readOnly
                  id='middle_name_id'
                  label='Отчество'
                  name='middle_name'
                  placeholder='Отчество'
                />
              </div>
            </Form>
          </div>
        </div>

        <TableComponent
          rowKey={(record) => record.id}
          loading={userVehicleInfoLoading}
          columns={columns}
          data={state}
          disabledButton
        />
      </div>
    </div>
  );
};

export default ProfileTechnique;
