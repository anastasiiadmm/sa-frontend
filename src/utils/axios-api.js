import axios from "axios";
import {apiURL} from "utils/config";

export const axiosApi = axios.create({
  baseURL: apiURL,
});