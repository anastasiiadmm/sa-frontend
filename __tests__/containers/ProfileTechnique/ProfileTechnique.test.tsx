import { BrowserRouter } from "react-router-dom";
import {cleanup } from "@testing-library/react";
import "../../../__mocks__/matchMedia.mock";
import "@testing-library/jest-dom";
import { mockedDispatch, mockedUseSelectors } from "../../../__mocks__/utils";

import renderer from "react-test-renderer";
import ProfileTechnique from "../../../src/containers/Technique/ProfileTechnique/ProfileTechnique";

afterEach(cleanup);

const data = {
    userVehicles: [
        {
            "id": 88,
            "readable_id": {
                "type": {},
                "key": null,
                "ref": null,
                "props": {
                    "title": "FieldThree",
                    "color": "#BBBBBB",
                    "overlayInnerStyle": {
                        "padding": "5px 15px",
                        "borderRadius": 15
                    },
                    "placement": "topRight",
                    "children": {
                        "type": "p",
                        "key": null,
                        "ref": null,
                        "props": {
                            "className": "text_hidden",
                            "children": "FieldThree"
                        },
                        "_owner": null,
                        "_store": {}
                    }
                },
                "_owner": null,
                "_store": {}
            },
            "tool": "None",
            "tool_width": "0",
            "tool_overlap": "0",
            "tool_width_total": "0",
            "left_right": "0.0",
            "area": "527.006"
        },
    ],
    userVehicleInfo: {
        results: [
            {
                "id": 88,
                "readable_id": "FieldThree",
                "tool": "None",
                "tool_width": "0",
                "tool_overlap": "0",
                "tool_width_total": "0",
                "left_right": "0.0",
                "area": "527.006"
            },
            {
                "id": 86,
                "readable_id": "send by almaz",
                "tool": "Борона 24 метра",
                "tool_width": "12",
                "tool_overlap": "-0.3",
                "tool_width_total": "11.7",
                "left_right": "0.0",
                "area": "927.006"
            }
        ],
        vehicle: {
            image: 'test/photos'
        }
    }
};


describe("<ProfileTechnique />", () => {
    test("Users table component should be in the document", async () => {
        mockedUseSelectors.mockReturnValue(data);
        const dispatch = jest.fn();
        mockedDispatch.mockReturnValue(dispatch);

        const tree = renderer
            .create(
                <BrowserRouter>
                    <ProfileTechnique />
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
