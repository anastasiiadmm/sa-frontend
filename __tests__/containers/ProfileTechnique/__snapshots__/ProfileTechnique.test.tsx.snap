// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileTechnique /> Users table component should be in the document 1`] = `
<div
  className="ProfileTechnique"
  data-testid="profile-technique-id"
>
  <div
    className="ProfileTechnique__table"
  >
    <div
      className="ProfileTechnique__header"
    >
      <div
        className="ProfileTechnique__header-title"
      >
        <button
          className="ProfileTechnique__back_handler"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="arrow"
            className="ProfileTechnique__arrow-left"
            src={
              {
                "process": [Function],
              }
            }
          />
        </button>
        <h3
          className="ant-typography ProfileTechnique__title css-dev-only-do-not-override-15rg2km"
          style={
            {
              "WebkitLineClamp": undefined,
            }
          }
        >
          Профиль техники - 
           -
           
           
        </h3>
      </div>
      <div>
        <a
          href="/open-map/undefined/local-tractor/undefined"
          onClick={[Function]}
        >
          <button
            className="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-link ant-btn-lg ProfileTechnique__open-map"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="ant-btn-icon"
            >
              <img
                alt="Техника на карте"
                src={
                  {
                    "process": [Function],
                  }
                }
                width={18}
              />
            </span>
            <span>
              Техника на карте
            </span>
          </button>
        </a>
      </div>
    </div>
    <div
      className="ProfileTechnique__technique-profile-info"
    >
      <div
        className="ant-image css-dev-only-do-not-override-15rg2km"
        style={
          {
            "height": undefined,
            "width": 242,
          }
        }
      >
        <img
          className="ant-image-img ProfileTechnique__technique-image"
          onLoad={[Function]}
          src="https://localhost:8000test/photos"
          style={
            {
              "borderRadius": 4,
              "height": undefined,
            }
          }
          width={242}
        />
      </div>
      <div
        className="ProfileTechnique__profile-info"
      >
        <form
          className="ant-form ant-form-vertical css-dev-only-do-not-override-15rg2km"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <h5
            className="ant-typography ProfileTechnique__profile-title css-dev-only-do-not-override-15rg2km"
            style={
              {
                "WebkitLineClamp": undefined,
              }
            }
          >
            Информация о технике
          </h5>
          <div
            className="ProfileTechnique__form-block"
          >
            <div
              className="ant-form-item css-dev-only-do-not-override-15rg2km"
            >
              <div
                className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
                style={{}}
              >
                <div
                  className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <label
                    className=""
                    htmlFor="description"
                    title="Название техники"
                  >
                    Название техники
                  </label>
                </div>
                <div
                  className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <input
                        className="ant-input ant-input-borderless css-dev-only-do-not-override-15rg2km input-styles undefined"
                        disabled={false}
                        id="description"
                        name="description"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Название техники"
                        readOnly={true}
                        style={{}}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ant-form-item css-dev-only-do-not-override-15rg2km"
            >
              <div
                className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
                style={{}}
              >
                <div
                  className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <label
                    className=""
                    htmlFor="license_plate"
                    title="Гос номер"
                  >
                    Гос номер
                  </label>
                </div>
                <div
                  className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <input
                        className="ant-input ant-input-borderless css-dev-only-do-not-override-15rg2km input-styles undefined"
                        disabled={false}
                        id="license_plate"
                        name="license_plate"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Гос номер"
                        readOnly={true}
                        style={{}}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ant-form-item css-dev-only-do-not-override-15rg2km"
            >
              <div
                className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
                style={{}}
              >
                <div
                  className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <label
                    className=""
                    htmlFor="vin"
                    title="VIN код"
                  >
                    VIN код
                  </label>
                </div>
                <div
                  className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <input
                        className="ant-input ant-input-borderless css-dev-only-do-not-override-15rg2km input-styles undefined"
                        disabled={false}
                        id="vin"
                        name="vin"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="VIN код"
                        readOnly={true}
                        style={{}}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <h5
            className="ant-typography ProfileTechnique__profile-title css-dev-only-do-not-override-15rg2km"
            style={
              {
                "WebkitLineClamp": undefined,
              }
            }
          >
            Информация о механизаторе
          </h5>
          <div
            className="ProfileTechnique__form-block"
          >
            <div
              className="ant-form-item css-dev-only-do-not-override-15rg2km"
            >
              <div
                className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
                style={{}}
              >
                <div
                  className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <label
                    className=""
                    htmlFor="operator_last_name"
                    title="Фамилия"
                  >
                    Фамилия
                  </label>
                </div>
                <div
                  className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <input
                        className="ant-input ant-input-borderless css-dev-only-do-not-override-15rg2km input-styles undefined"
                        disabled={false}
                        id="operator_last_name"
                        name={
                          [
                            "operator",
                            "last_name",
                          ]
                        }
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Фамилия"
                        readOnly={true}
                        style={{}}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ant-form-item css-dev-only-do-not-override-15rg2km"
            >
              <div
                className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
                style={{}}
              >
                <div
                  className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <label
                    className=""
                    htmlFor="operator_first_name"
                    title="Имя"
                  >
                    Имя
                  </label>
                </div>
                <div
                  className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <input
                        className="ant-input ant-input-borderless css-dev-only-do-not-override-15rg2km input-styles undefined"
                        disabled={false}
                        id="operator_first_name"
                        name={
                          [
                            "operator",
                            "first_name",
                          ]
                        }
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Имя"
                        readOnly={true}
                        style={{}}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ant-form-item css-dev-only-do-not-override-15rg2km"
            >
              <div
                className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
                style={{}}
              >
                <div
                  className="ant-col ant-form-item-label css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <label
                    className=""
                    htmlFor="operator_middle_name"
                    title="Отчество"
                  >
                    Отчество
                  </label>
                </div>
                <div
                  className="ant-col ant-form-item-control css-dev-only-do-not-override-15rg2km"
                  style={{}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <input
                        className="ant-input ant-input-borderless css-dev-only-do-not-override-15rg2km input-styles undefined"
                        disabled={false}
                        id="operator_middle_name"
                        name={
                          [
                            "operator",
                            "middle_name",
                          ]
                        }
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Отчество"
                        readOnly={true}
                        style={{}}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      className="ant-table-wrapper css-dev-only-do-not-override-15rg2km"
    >
      <div
        className="ant-spin-nested-loading css-dev-only-do-not-override-15rg2km"
      >
        <div
          className="ant-spin-container"
        >
          <div
            className="ant-table ant-table-empty ant-table-scroll-horizontal"
          >
            <div
              className="ant-table-container"
            >
              <div
                className="ant-table-content"
                onScroll={[Function]}
                style={
                  {
                    "overflowX": "auto",
                    "overflowY": "hidden",
                  }
                }
              >
                <table
                  style={
                    {
                      "minWidth": "100%",
                      "tableLayout": "auto",
                      "width": 950,
                    }
                  }
                >
                  <colgroup>
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "18%",
                        }
                      }
                    />
                  </colgroup>
                  <thead
                    className="ant-table-thead"
                  >
                    <tr>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Задача на обработку
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Инструмент
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Обрабатываемая ширина, м
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Ширина перекрытия, м
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Итоговая ширина, м
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Смещение, м
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Обрабатываемая площадь, га
                      </th>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope={null}
                        style={{}}
                      />
                    </tr>
                  </thead>
                  <tbody
                    className="ant-table-tbody"
                  >
                    <tr
                      aria-hidden="true"
                      className="ant-table-measure-row"
                      style={
                        {
                          "fontSize": 0,
                          "height": 0,
                        }
                      }
                    >
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                    </tr>
                    <tr
                      className="ant-table-placeholder"
                      style={
                        {
                          "display": null,
                        }
                      }
                    >
                      <td
                        className="ant-table-cell"
                        colSpan={8}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        style={{}}
                      >
                        <div
                          className="ant-table-expanded-row-fixed"
                          style={
                            {
                              "left": 0,
                              "overflow": "hidden",
                              "position": "sticky",
                              "width": 0,
                            }
                          }
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pagination"
    >
      <div
        className="pagination__pagination_show"
      >
        <div
          className="glav_pagination"
        >
          <div
            className="next_table"
          >
            <button
              className="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-prev"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              <span
                className="ant-btn-icon"
              >
                <img
                  alt="backIcons"
                  src={
                    {
                      "process": [Function],
                    }
                  }
                />
              </span>
            </button>
            <button
              className="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-next"
              disabled={true}
              onClick={[Function]}
              role="button"
              type="button"
            >
              <span
                className="ant-btn-icon"
              >
                <img
                  alt="nextIcons"
                  src={
                    {
                      "process": [Function],
                    }
                  }
                />
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
